packages/lodestar/src/cache/abstract.ts

```diff
@@ -0,0 +1,68 @@
+import {AnySSZType} from "@chainsafe/ssz-type-schema";
+import {CacheId, CacheIdFunction} from "./types";
+import {clone, hashTreeRoot} from "@chainsafe/ssz";
+
+export abstract class CacheItem<T> {
+
+  protected sszType: AnySSZType;
+
+  protected idFunction: CacheIdFunction<T>;
+
+  protected cache: Map<CacheId, T> = new Map();
```

Comment:
could we abstract this further? What if we wanted to use Redis ?

Reply:
Can we really package redis with lodestar? I would leave further abstraction until we have some use case, should be fairly easy to abstract more

Reply:
no idea - just a thought

Reply:
Isn't it just installing redis npm package and then it's ready to use? It's in-memory so should be lightweight, it also supports expiring feature.

Reply:
if we support expiring feature then probably we'll end up with implement something like redis.

Reply:
I'm pretty sure thats just redis client. You need redis server for client to work. There are redis mocks, but they are a bit bloated and not maintained